src/6model/reprs/VMArray.c

```diff
@@ -877,7 +891,7 @@ static void aslice(MVMThreadContext *tc, MVMSTable *st, MVMObject *src, void *da
 
     elems = end - start + 1;
     if (d_repr_data) {
-        set_size_internal(tc, d_body, elems, d_repr_data);
+        set_size_internal(tc, src, d_body, elems, d_repr_data);
```

Comment:
could this perhaps need dest rather than source?
```suggestion
        set_size_internal(tc, dest, d_body, elems, d_repr_data);
```